### PR TITLE
Allow case-insensitive animal_project_code, network_project_code, station_name

### PR DESCRIPTION
**Pull request by peterdesmet**
Implements #140

- [x] Casts those input parameter to lowercase in functions that use them
- [x] ~Updates `list_` functions so it only returns lowercase values~: they return the values as is (so the functions are unchanged), but the `check_value()` is case-insensitive and the error message will return lowercase values:
- [x] Make SQL in `get_` queries case-insensitive
- [x] Add tests

### CODE REVIEW

**Comment by peterdesmet** [comment]
@damianooldoni this is now implemented.

`list_` functions are still the same:

```
list_animal_project_codes()
 [1] "2010_phd_reubens"          "2010_phd_reubens_sync"     "2011_rivierprik"          
 [4] "2012_leopoldkanaal"        "2013_albertkanaal"         "2014_demer"               
 [7] "2015_Albertkanaal_VPS_Ham"
```

`get_` functions work now independent of case. Values in the returned df are as they are in the database:

```
> get_animals(con, animal_project_code = "2015_AlbertKANaal_vps_HAM")
# A tibble: 13 x 64
      pk animal_id animal_project_… tag_id tag_fk scientific_name common_name aphia_id animal_label
   <int>     <int> <chr>            <chr>  <chr>  <chr>           <chr>          <int> <chr>       
 1   650       650 2015_Albertkana… A69-1… 672    Sync tag        NA                NA NA          
```

Incorrect values still give an error, but the returned values in error message are lowercase (difficult to get around that, but fine I think):

```
get_animals(con, animal_project_code = "AlbertKANaal_vps_HAM")
 Error: Invalid value(s) for animal_project_code argument.
Valid inputs are: 2010_phd_reubens, 2010_phd_reubens_sync, 2011_rivierprik, 2012_leopoldkanaal, 2013_albertkanaal, 2014_demer, 2015_albertkanaal_vps_ham, 2015_dijle, 2015_fint, 2015_homarus, 2015_phd_verhelst_cod, 2015_phd_verhelst_eel, 2016_albertkanaal_vps_ham, 2016_phd_vergeynst, 2017_albertkanaal_vps_ham, 2017_iller, 2018_ec, 2018_iller, 2019_grotenete, adst-shark and others... 
```

Let me know if we should add tests for this too.

**Review by damianooldoni — CHANGES_REQUESTED** [review]
Nice work @peterdesmet!

I think it is only needed an improvement of documentation in section @example

**Comment by damianooldoni** [comment]
@peterdesmet: I added tests but for get_animals() I cannot really check it due to the fact I don't have access to all projects. Could you run the tests, please? Thanks.

**Comment by damianooldoni** [comment]
Now I have access to all projects, I could check the tests for get_animals() and they passed as expected.

I also changed the animal project used for testing `download_dataset()` as dataset output from  `2015_homarus` is changing over time quite often and I got again a new test failure. Changed to `2014_demer`, which is a closed project, so stable for testing.

@peterdesmet : to add to my previous review: would you plan to update package version? This new functionality deserves it, I think :+1:

**Comment by peterdesmet** [comment]
You can bump to version number `0.1.0.9140`, where not quite at 0.2 yet

File: R/get_animals.R [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-#' Get data for animals, with options to filter results. Associated tag 
-#' information is available in columns starting with `tag`. If multiple tags 
+#' Get data for animals, with options to filter results. Associated tag
+#' information is available in columns starting with `tag`. If multiple tags
 #' are associated with a single animal, the information is comma-separated.
 #'
 #' @param connection A connection to the ETN database. Defaults to `con`.
 #' @param animal_id Integer (vector). One or more animal ids.
 #' @param animal_project_code Character (vector). One or more animal projects.
+#'   Case-insensitive.
```
[the remark this answers is not in the document]
Comment (peterdesmet) [reply R/get_animals.R:10]: Would have to do this for all functions I guess?

[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/get_animals.R:10]: Yes, I think so. :smile:

[the remark this answers is not in the document]
Comment (peterdesmet) [reply R/get_animals.R:10]: Will be addressed in #147

**Review by damianooldoni — APPROVED** [review]
Great work

**Comment by damianooldoni** [comment]
Merge @peterdesmet, please :+1:

**Pull request merged**